pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -1459,8 +1460,18 @@ static void updateColumnNames(String rawTableName, PinotQuery pinotQuery, boolea
       Map<String, String> columnNameMap) {
     Map<String, String> aliasMap = new HashMap<>();
     if (pinotQuery != null) {
+      Expression selectStarExpr = null;
```

Comment:
No need to store and pass this expression. This can be changed to a boolean field `hasSelectStar`. We can add a constant for `*` identifier, and the compare can be simplified to `expression.equals(STAR)`

Reply:
Addressed.